packages/portfolio-contract/src/resolver/resolver.exo.ts

```diff
@@ -22,36 +21,45 @@ import { E } from '@endo/far';
 import { M } from '@endo/patterns';
 import { TxStatus, TxType } from './constants.js';
 import type {
-  CCTPSettlementArgs,
-  CCTPSettlementOfferArgs,
-  CCTPTransactionKey,
   PublishedTx,
+  TransactionKey,
+  TransactionSettlementArgs,
+  TransactionSettlementOfferArgs,
+} from './types.js';
+import {
+  ResolverOfferArgsShapes,
+  TransactionSettlementArgsShape,
 } from './types.js';
-import { CCTPSettlementArgsShape, ResolverOfferArgsShapes } from './types.js';
 
-type CCTPTransactionEntry = {
+type TransactionEntry = {
   destinationAddress: AccountId;
   amountValue: bigint;
   vowKit: VowKit<void>;
 };
 
 const trace = makeTracer('Resolver');
 
-const ClientFacetI = M.interface('CCTPResolverClient', {
-  registerCCTPTransaction: M.call(M.string(), M.nat()).returns(VowShape),
+const ClientFacetI = M.interface('ResolverClient', {
+  registerTransaction: M.call(M.string(), M.string(), M.nat()).returns(
```

Comment:
The static type of the 1st arg is not `string`, it's `TxType`.

```suggestion
  registerTransaction: M.call(TxTypeShape, M.string(), M.nat()).returns(
```

where `TxTypeShape` is something like `M.or(...Object.keys(TxType))`.